tests/test_config.py

```diff
@@ -51,3 +51,112 @@ def test_no_ini_path(self):
 
         with self.assertRaises(ValueError):
             parser.parse_args([])
+
+
+class TestGetConfigValue(unittest.TestCase):
+    """Tests for caramel.config._get_config_value"""
+
+    def test_argument_only(self):
```

Comment:
Missing here is a testcase where you're not running "setattr"  on arguments first.   Is that a branch of the tests that need to happen?

Reply:
Good catch! There may be config variables that might not be available as cl arguments, therefore not being set as a attribute in the argparse.Namespace object, but you still want to check the environment or settings file. Will fix immediately